clang/test/CIR/CodeGenBuiltins/X86/avx512bw-builtins.c

```diff
@@ -768,3 +768,47 @@ unsigned char test_ktestz_mask64_u8(__mmask64 A, __mmask64 B) {
   return _ktestz_mask64_u8(A, B);
 }
 
+
+
+__m512i test_mm512_movm_epi16(__mmask32 __A) {
+  // CIR-LABEL: _mm512_movm_epi16
+  // CIR: %{{.*}} = cir.cast bitcast %{{.*}} : !u32i -> !cir.vector<!cir.int<s, 1> x 32>
+  // CIR: %{{.*}} = cir.cast integral %{{.*}} : !cir.vector<!cir.int<s, 1> x 32> -> !cir.vector<!s16i x 32>
+  
+  // LLVM-LABEL: @test_mm512_movm_epi16
+  // LLVM:  %{{.*}} = bitcast i32 %{{.*}} to <32 x i1>
+  // LLVM:  %{{.*}} = sext <32 x i1> %{{.*}} to <32 x i16>
+
+  // OGCG-LABEL: @test_mm512_movm_epi16
+  // OGCG:  %{{.*}} = bitcast i32 %{{.*}} to <32 x i1>
+  // OGCG:  %{{.*}} = sext <32 x i1> %{{.*}} to <32 x i16>
+  return _mm512_movm_epi16(__A); 
+}
+
+__mmask64 test_mm512_movepi8_mask(__m512i __A) {
+  // CIR-LABEL: @_mm512_movepi8_mask
+  // CIR: %{{.*}} = cir.vec.cmp(lt, %{{.*}}, %{{.*}}) : !cir.vector<{{!s8i|!u8i}} x 64>, !cir.vector<!cir.int<u, 1> x 64>
```

Comment:
This also produces a bitcast from <32 x i1> to i32. Can you add that to the checks (CIR, LLVM, and OGCG) on all the relevant cases? I realize the classic codegen equivalent of this test only checks the icmp instruction, but it should be checking the bitcast too.